lib/node_modules/@stdlib/lapack/base/dlange/lib/base.js

```diff
@@ -0,0 +1,362 @@
+		}
+		ia += da1;
+	}
+	return value;
+}
+
+/**
+* Returns the value of the infinity norm of a real matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @param {Float64Array} work - work array, should have `M` indexed elements if column-major layout is used
+* @param {integer} strideWork - stride length of `work`
+* @param {NonNegativeInteger} offsetWork - starting index of `work`
+* @returns {number} required norm value
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+* var work = new Float64Array( 3 );
+*
+* var out = infinityNorm( 3, 4, A, 4, 1, 0, work, 1, 0 );
+* // returns 30.0
+*/
+function infinityNorm( M, N, A, strideA1, strideA2, offsetA, work, strideWork, offsetWork ) { // eslint-disable-line max-len
+	var value;
+	var temp;
+	var sum;
+	var ia1;
+	var ia2;
+	var iw;
+	var i;
+	var j;
+
+	if ( isRowMajor( [ strideA1, strideA2 ] ) ) {
+		value = 0.0;
+		ia1 = offsetA;
+		for ( j = 0; j < M; j++ ) {
+			sum = dasum( N, A, strideA2, ia1 );
+			if ( value < sum || isnan( sum ) ) {
+				value = sum;
+			}
+			ia1 += strideA1;
+		}
+	} else {
+		iw = offsetWork;
+		for ( i = 0; i < M; i++ ) {
+			work[ iw ] = 0.0;
+			iw += strideWork;
+		}
+
+		ia1 = offsetA;
+		for ( j = 0; j < N; j++ ) {
+			ia2 = 0;
+			iw = offsetWork;
+			for ( i = 0; i < M; i++ ) {
+				work[ iw ] += abs( A[ ia1 + ia2 ] );
+				iw += strideWork;
+				ia2 += strideA1;
+			}
+			ia1 += strideA2;
+		}
+
+		value = 0.0;
+
+		iw = offsetWork;
+		for ( i = 0; i < M; i++ ) {
+			temp = work[ iw ];
+			if ( value < temp || isnan( temp ) ) {
+				value = temp;
+			}
+			iw += strideWork;
+		}
+	}
+	return value;
+}
+
+/**
+* Returns the absolute value of the frobenius norm of a real matrix `A`.
+*
+* @private
+* @param {NonNegativeInteger} M - number of rows in `A`
+* @param {NonNegativeInteger} N - number of columns in `A`
+* @param {Float64Array} A - input array
+* @param {integer} strideA1 - stride of the first dimension of `A`
+* @param {integer} strideA2 - stride of the second dimension of `A`
+* @param {NonNegativeInteger} offsetA - starting index of `A`
+* @returns {number} required norm value
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var A = new Float64Array( [ 1.0, 4.0, 7.0, 10.0, 2.0, 5.0, 8.0, 11.0, 3.0, 6.0, 9.0, 12.0 ] );
+*
+* var out = frobeniusNorm( 3, 4, A, 4, 1, 0 );
+* // returns ~25.5
+*/
+function frobeniusNorm( M, N, A, strideA1, strideA2, offsetA ) {
+	var out;
+	var da0;
+	var da1;
+	var S1;
+	var S2;
+	var ia;
+	var i;
+
+	out = new Float64Array( [ 0.0, 1.0 ] );
```

Comment:
This can be moved to the parent scope as a workspace which can be reused below. This avoids repeated allocation.